Rubberduck.CodeAnalysis/Inspections/Concrete/ImplicitlyTypedConstInspection.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Resources.Inspections;
+
+namespace Rubberduck.CodeAnalysis.Inspections.Concrete
+{
+    public sealed class ImplicitlyTypedConstInspection : InspectionBase
+    {
+        public ImplicitlyTypedConstInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            var declarationFinder = State.DeclarationFinder;
+
+            var implicitlyTypedConsts = declarationFinder.UserDeclarations(DeclarationType.Constant)
+                .Where(declaration => !declaration.IsTypeSpecified);
+
+            return implicitlyTypedConsts.Select(Result);
+        }
+
+        private IInspectionResult Result(Declaration declaration)
```

Comment:
Not for this PR, but I like this.. @MDoerner any thoughts on having `protected virtual IInspectionResult` overloads in the base class? I feel like the description string could be templated, too.

Reply:
You might want to have a look at my latest PR.